gui/src/2D/advancedDynamicTexture.ts

```diff
@@ -1154,15 +1165,24 @@ export class AdvancedDynamicTexture extends DynamicTexture {
      * @param foreground defines a boolean indicating if the texture must be rendered in foreground (default is true)
      * @param scene defines the hosting scene
      * @param sampling defines the texture sampling mode (Texture.BILINEAR_SAMPLINGMODE by default)
+     * @param adaptiveScaling defines whether to automatically scale root to match hardwarescaling (false by default)
      * @returns a new AdvancedDynamicTexture
      */
-    public static CreateFullscreenUI(name: string, foreground: boolean = true, scene: Nullable<Scene> = null, sampling = Texture.BILINEAR_SAMPLINGMODE): AdvancedDynamicTexture {
+    public static CreateFullscreenUI(name: string, foreground: boolean = true, scene: Nullable<Scene> = null, sampling = Texture.BILINEAR_SAMPLINGMODE, adaptiveScaling : boolean = false): AdvancedDynamicTexture {
         var result = new AdvancedDynamicTexture(name, 0, 0, scene, false, sampling);
         // Display
         var layer = new Layer(name + "_layer", null, scene, !foreground);
         layer.texture = result;
         result._layerToDispose = layer;
         result._isFullscreen = true;
+
+        const resultScene = result.getScene();
```

Comment:
scene is already defined, is that a different scene? can it be a different scene?

Reply:
This is that the user can pass in and it can be null. If it is null the layer constructor takes care of it with. 
 this._scene = <Scene>(scene || EngineStore.LastCreatedScene);

Reply:
I understand that this will (probably) never fail, but it is prone to misunderstandings without diving into the code.

You pass a scene object that can be null, you use it to create an ADT, then you use it to create a layer, and then you are taking the scene from the ADT and use it (to get the engine).
You can either say - "the scene is `(scene || EngineStore.LastCreatedScene);`", or initially use the scene from the ADT to also create the layer. This way you are staying clear and consistent. Option 2 is the better one :-)